Bundle/CoreBundle/Resources/script/ngApp/directives/widget.js

```diff
@@ -0,0 +1,69 @@
+angular.module('ngApp').directive('widget', function ($compile, $rootScope) {
+    return {
+        restrict:'A',
+        scope: {
+            widgetMap: '='
+        },
+        link: function(scope, element, attrs) {
+            scope.positions = [];
+            var renderButton = function () {
+                if (element.prev().attr('position') && element.prev().attr('widget-map') == element.attr('widget-map')) {
+                    element.prev().remove();
+                }
+                if (element.next().attr('position') && element.next().attr('widget-map') == element.attr('widget-map')) {
+                    element.next().remove();
+                }
+
+                widgets = $vic('.vic-widget-container', '#vic-slot-' + scope.$parent.slotId);
+                if (!("max" in scope.$parent.options) || ("max" in scope.$parent.options) && scope.$parent.options.max > widgets.length) {
+                    if (scope.positions.before == 1) {
```

Comment:
please refactor these two conditions, it can be squashed easy, there is only one different variable